streamtrace.cc

```diff
@@ -1695,6 +1695,51 @@ void decode_cap(capability_register &cap, uint64_t val2, uint64_t val3,
 	cap.length = val5;
 	cap.offset = val3 - cap.base;
 }
+/**
+ * Decode a 128-bit capability register from streamtrace values.  `val2` is always
+ * `val2` from the streamtrace, but `val1` is either `val1` or `pc` depending
+ * on the trace format. val1/pc is the 1st half of the capability containing the
+ * pointer and val2 is the 2nd half of the compressed capability containing the
+ * bounds and permissions. 
+ */
```

Comment:
Yes.  This is brought forward from the older variations of this function.  If we want to change the comment, we should change it everywhere.